airflow/models/taskinstance.py

```diff
@@ -509,14 +509,8 @@ class TaskInstance(Base, LoggingMixin):
         viewonly=True,
     )
 
-    trigger = relationship(
-        "Trigger",
-        primaryjoin="TaskInstance.trigger_id == Trigger.id",
-        foreign_keys=trigger_id,
-        uselist=False,
-        innerjoin=True,
-    )
-
+    trigger = relationship("Trigger", uselist=False)
```

Comment:
Why don't we need innerjoin any longer?

Reply:
i don't think we ever needed it.  not every TI has a trigger, so what happens is... when you reference this property in a query, the TIs that don't have a trigger drop out of the result set.
i looked, and it seems that there was only [one usage](https://github.com/apache/airflow/blob/aa63eb969320cacadaed04fd54c716f57da7a2c7/airflow/www/views.py#L1557) of this attr, and it it, they actually overrided that setting, to set it to false...